src/plugins/menu/src/lib.rs

```diff
@@ -222,25 +227,72 @@ where
 		let play = GameState::Play;
 
 		app.add_ui::<UIOverlay, TLocalization::TLocalizationServer, TGraphics::TUiCamera>(play)
+			.add_observer(QuickbarPanel::add_quickbar_primer::<TSettings::TKeyMap<PlayerSlot>>)
 			.add_systems(
 				Update,
 				(
-					QuickbarPanel::add_quickbar_primer::<TSettings::TKeyMap<PlayerSlot>>,
+					QuickbarPanel::set_icon::<TPlayers::TPlayer, TLoadout::TSlots>,
+					QuickbarPanel::set_color::<
+						TPlayers::TPlayer,
+						TSettings::TKeyMap<PlayerSlot>,
+						TLoadout::TSlots,
+					>,
 					panel_colors::<QuickbarPanel>,
 				)
 					.run_if(in_state(play)),
 			);
 	}
 
 	fn combo_overview(&self, app: &mut App) {
-		app.add_dropdown::<TLocalization::TLocalizationServer, KeySelect<AppendSkill>>();
+		type InVerticalDropdown<TSkill> = ComboSkillButton<DropdownItem<Vertical>, TSkill>;
+		type InHorizontalDropdown<TSkill> = ComboSkillButton<DropdownItem<Horizontal>, TSkill>;
+		type DropdownButton<TSkill> = ComboSkillButton<DropdownTrigger, TSkill>;
+
+		let combo_overview = GameState::IngameMenu(MenuState::ComboOverview);
+
+		app
+			.add_ui::<ComboOverview<TLoadout::TSkill>, TLocalization::TLocalizationServer, TGraphics::TUiCamera>(combo_overview)
+			.add_dropdown::<TLocalization::TLocalizationServer, KeySelect<AppendSkill>>()
+			.add_dropdown::<TLocalization::TLocalizationServer, InVerticalDropdown<TLoadout::TSkill>>()
+			.add_dropdown::<TLocalization::TLocalizationServer, InHorizontalDropdown<TLoadout::TSkill>>()
+			.add_systems(
+				Update,
+				(
+					ComboOverview::<TLoadout::TSkill>::update_from::<TPlayers::TPlayer, TLoadout::TCombos>,
+					KeySelectDropdownCommand::insert_dropdown::<TPlayers::TPlayer, TLoadout::TCombos>,
+					SkillSelectDropdownCommand::<Vertical>::insert_dropdown::<TPlayers::TPlayer, TLoadout::TSlots, TLoadout::TSkills>,
+					SkillSelectDropdownCommand::<Horizontal>::insert_dropdown::<TPlayers::TPlayer, TLoadout::TSlots, TLoadout::TSkills>,
+					InVerticalDropdown::<TLoadout::TSkill>::update::<TPlayers::TPlayer, TLoadout::TCombos>,
+					InHorizontalDropdown::<TLoadout::TSkill>::update::<TPlayers::TPlayer, TLoadout::TCombos>,
+					DeleteSkill::from_combos::<TPlayers::TPlayer, TLoadout::TCombos>,
+					DropdownButton::<TLoadout::TSkill>::visualize_invalid::<Unusable, TPlayers::TPlayer, TLoadout::TSlots>,
+				)
+					.chain()
+					.run_if(in_state(combo_overview)),
+			);
 	}
```

Comment:
Throw this into a loose function over the target types. This might make it more readable and format-able.

Reply:
a loose function requires a million constraints on that functions. However, by less chaining on `app` we can regain auto formatting features.